tools/azure-sdk-tools/devtools_testutils/keyvault_testcase.py

```diff
@@ -0,0 +1,129 @@
+        self.location = location
+        self.sku = sku
+        self.permissions = permissions
+        self.enabled_for_deployment = enabled_for_deployment
+        self.enabled_for_disk_encryption = enabled_for_disk_encryption
+        self.enabled_for_template_deployment = enabled_for_template_deployment
+        self.enable_soft_delete = enable_soft_delete
+        self.resource_group_parameter_name = resource_group_parameter_name
+        self.parameter_name = parameter_name
+        if random_name_enabled:
+            self.resource_moniker = self.name_prefix
+        self.client_oid = None
+
+    def create_resource(self, name, **kwargs):
+        self.client_oid = self.test_class_instance.set_value_to_scrub("CLIENT_OID", CLIENT_OID)
+        if self.is_live:
+            group = self._get_resource_group(**kwargs).name
+            access_policies = [
+                AccessPolicyEntry(
+                    tenant_id=self.test_class_instance.get_settings_value("TENANT_ID"),
+                    object_id=self.client_oid,
+                    permissions=self.permissions,
+                )
+            ]
+            properties = VaultProperties(
+                tenant_id=self.test_class_instance.get_settings_value("TENANT_ID"),
+                sku=Sku(name=self.sku),
+                access_policies=access_policies,
+                vault_uri=None,
+                enabled_for_deployment=self.enabled_for_deployment,
+                enabled_for_disk_encryption=self.enabled_for_disk_encryption,
+                enabled_for_template_deployment=self.enabled_for_template_deployment,
+                enable_soft_delete=self.enable_soft_delete,
+                enable_purge_protection=None,
+            )
+            parameters = VaultCreateOrUpdateParameters(location=self.location, properties=properties)
+            self.client = self.create_mgmt_client(KeyVaultManagementClient)
+
+            # ARM may return not found at first even though the resource group has been created
+            retries = 4
+            for i in range(retries):
+                try:
+                    vault = self.client.vaults.create_or_update(group, name, parameters).result()
+                except Exception as ex:
+                    if "ResourceGroupNotFound" not in str(ex) or i == retries - 1:
+                        raise
+                    time.sleep(3)
+            self.test_class_instance.scrubber.register_name_pair(
+                name,
+                self.resource_moniker
+            )
+            vault_uri = vault.properties.vault_uri
+        else:
+            # playback => we need only the uri used in the recording
+            vault_uri = "https://{}.vault.azure.net/".format(name)
+        return {self.parameter_name: vault_uri}
+
+    def remove_resource(self, name, **kwargs):
+        if self.is_live:
+            group = self._get_resource_group(**kwargs).name
```

Comment:
```py
def remove_resource(self, name, **kwargs):
  self._get_resource_group(**kwargs)

def _get_resource_group(self, **kwargs):
  return kwargs[...]
```
So `remove_resource` already has what it needs?

Reply:
keeping bc _get_resource_group is called with purpose in creating a keyvault